site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -342,6 +342,168 @@ If you add it to your own file, you should write something like
+and just using emscripten these issues probably won't come up but
+if you're an experienced JavaScript programmer you need to be aware
+many common JavaScript practices can not be used in emscripten
+library files.
+
+To save space, by default, emscripten only includes library properties
+referenced from C/C++. It does this by calling ``toString`` on each 
+used property. Conceptually you can consider it like this
+
+.. code-block:: javascript
+
+   usedLibraryFunctions.forEach(function(name) {
+      var func = LibraryManager.library[name];
+      emitCode("function _" + name + func.toString().replace("function", ""));
+   });
+   
+It's actually not that simple as it will emit JavaScript objects
+but any functions they reference must be expressible by calling `toString`.
+   
+The implications of that are as follows
+
+-  You can not use ES6 object shorthand function syntax.
+-  You can not reference/bind other functions directly.
+-  You can not use a closure directly.
+-  You can not use ES6 arrow syntax.
+-  You can not use other function transformations like currying.
+
+.. code-block:: javascript
+
+   mergeInto(LibraryManager.library, {
+     // bad: ES6 object shorthand function syntax not allowwed
+     bad_01() {
+       alert('hi');
+     },
+     
+     // bad: Can not bind/reference other functions directly.
+     bad_02: document.querySelector.bind(document),
+     
+     // bad: You can not use a closure directly
+     bad_03: (function() {
+       var callCount = 0;
+       return function() {
+         console.log("times called: ", ++callCount);
+       };
+     }()),
+     
+     // bad: You can not use ES6 arrow syntax
+     bad_04: () => { console.log("hi!"); },
+     
+     // bad: You can not curry/transform
+     bad_05: curry(scrollTo, 0),  
+   });
+   
+Workarounds to the above issues
+
+You can put code in another file using a ``<script>`` tag 
+OR ``--pre-js`` OR ``--post-js``. Of course many of these
+features work deeper in a function but the function directly 
+attached to the object passed to ``mergeInto`` must be a plain old 
+vanilla JavaScript ES5 function.
+
+One way to workaround some of these issues is to have a function
+replace itself and have it called during initialization.
+
+.. code-block:: javascript
+
+   mergeInto(LibraryManager.library, {
+
+     // Can not bind/reference other functions directly workround
+     good_02__postset: '_good_02();',
+     good_02: function() {
+       _good_o2 = document.querySelector.bind(document);
+     },
+     
+     // Can not use a closure workaround
+     good_03__postset: '_good_03();',
+     good_03: function() {
+       var callCount = 0;
+       _good_03 = function() {
+         console.log("times called: ", ++callCount);
+       };
+     },
+     
+     // Can not curry/transform workaround
+     good_05__postset: '_good_05();',
+     good_05: function() {
+       _good_05 = curry(scrollTo, 0);  
+    },
+
+   });
+
```

Comment:
I would add a comment here "(A `__postset` is a little code that the compiler will emit directly in the output file)".